docs/api-docs/analytics/bodl-overview.md

```diff
@@ -0,0 +1,427 @@
+    {{inject "cartId" cart_id}}
+    // https://developer.bigcommerce.com/theme-objects/cart
+    // (Fetching selective cart data to prevent additional payment button object html from causing JS parse error)
+    {{inject "cartItemAdded" cart.added_item}}
+  }
+  var BODL = JSON.parse({{jsContext}});
+
+  if (BODL.categoryName) {
+    BODL.category = {
+      name: BODL.categoryName,
+      products: BODL.categoryProducts,
+    }
+  }
+
+  if (BODL.productTitle) {
+    BODL.product = {
+      id: BODL.productId,
+      title: BODL.productTitle,
+      price: {
+        without_tax: {
+          currency: BODL.productCurrency,
+          value: BODL.productPrice,
+        },
+      },
+    }
+  }
+
+  BODL.getCartItemContentId = (item) => {
+    switch(item.type) {
+      case 'GiftCertificate':
+        return item.type;
+      default:
+        return item.product_id;
+    }
+  }
+
+  BODL.getQueryParamValue = (name) => {
+    var cleanName = name.replace(/[\[]/, '\\[').replace(/[\]]/, '\\]');
+    var cleanRegex = new RegExp('[\\?&]' + cleanName + '=([^&#]*)');
+    var results = cleanRegex.exec(window.location.search);
+    return results === null ? '' : decodeURIComponent(results[1].replace(/\+/g, ' '));
+  }
+
+  // Replace the following example with an analytics instantiation script for your engine of choice.
+  !function (w, d, t) {
+    w.SampleAnalyticsObject=t;var sampleAnalyticsProvider=w[t]=w[t]||[];sampleAnalyticsProvider.methods=["page","track","identify","instances","debug","on","off","once","ready","alias","group","enableCookie","disableCookie"],sampleAnalyticsProvider.setAndDefer=function(t,e){t[e]=function(){t.push([e].concat(Array.prototype.slice.call(arguments,0)))}};for(var i=0;i<sampleAnalyticsProvider.methods.length;i++)sampleAnalyticsProvider.setAndDefer(sampleAnalyticsProvider,sampleAnalyticsProvider.methods[i]);sampleAnalyticsProvider.instance=function(t){for(var e=sampleAnalyticsProvider._i[t]||[],n=0;n<sampleAnalyticsProvider.methods.length;n++)sampleAnalyticsProvider.setAndDefer(e,sampleAnalyticsProvider.methods[n]);return e},sampleAnalyticsProvider.load=function(e,n){var i="https://analytics.sample.com/i18n/pixel/events.js";sampleAnalyticsProvider._i=sampleAnalyticsProvider._i||{},sampleAnalyticsProvider._i[e]=[],sampleAnalyticsProvider._i[e]._u=i,sampleAnalyticsProvider._t=sampleAnalyticsProvider._t||{},sampleAnalyticsProvider._t[e]=+new Date,sampleAnalyticsProvider._o=sampleAnalyticsProvider._o||{},sampleAnalyticsProvider._o[e]=n||{},sampleAnalyticsProvider._partner=sampleAnalyticsProvider._partner||"BigCommerce";var o=document.createElement("script");o.type="text/javascript",o.async=!0,o.src=i+"?sdkid="+e+"&lib="+t;var a=document.getElementsByTagName("script")[0];a.parentNode.insertBefore(o,a)};
+
+    sampleAnalyticsProvider.load('<%= property_id %>');
+    sampleAnalyticsProvider.page();
+
+    // Advanced Matching
+    if (BODL.customer && BODL.customer.id) {
```

Comment:
@becomevocal @bc-derrekpearson as far as I can see, we never create BODL.customer, so none of this customer data will ever make it into BODL. This will have failed silently when @traci707 ran the script in the console. Am I missing something?